internal/wsjs/wsjs.go

```diff
@@ -57,29 +59,33 @@ type WebSocket struct {
 	v js.Value
 }
 
-func (c *WebSocket) setBinaryType(typ string) {
+func (c WebSocket) setBinaryType(typ string) {
 	c.v.Set("binaryType", string(typ))
 }
 
-func (c *WebSocket) BufferedAmount() uint32 {
-	return uint32(c.v.Get("bufferedAmount").Int())
-}
-
-func (c *WebSocket) addEventListener(eventType string, fn func(e js.Value)) {
-	c.v.Call("addEventListener", eventType, js.FuncOf(func(this js.Value, args []js.Value) interface{} {
+func (c WebSocket) addEventListener(eventType string, fn func(e js.Value)) func() {
+	f := js.FuncOf(func(this js.Value, args []js.Value) interface{} {
 		fn(args[0])
```

Comment:
It's really important to [not block inside of a `js.Func` wrapped function](https://golang.org/pkg/syscall/js/#FuncOf). Doing so will block the JavaScript event loop and prevent any other JavaScript functions from running. You should almost always start a new goroutine inside of `js.FuncOf`.

For example:

```go
	f := js.FuncOf(func(this js.Value, args []js.Value) interface{} {
		go fn(args[0])
		return nil
	})
```

Reply:
Nice catch, fixed.